stan/math/prim/scal/prob/normal_rng.hpp

```diff
@@ -1,39 +1,66 @@
 #ifndef STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 #define STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 
-#include <boost/random/normal_distribution.hpp>
-#include <boost/random/variate_generator.hpp>
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/err/check_positive.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/err/check_positive_finite.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    normal_rng(double mu,
-               double sigma,
+    /**
```

Comment:
A lot of the docs are missing for the RNGs. I've just been adding stuff. It's a lot of boilerplate generic stuff but it might as well be there in case anyone is curious what these things do at the C++ level.